torch/jit/__init__.py

```diff
@@ -789,6 +790,95 @@ def wrap_check_inputs(check_inputs):
 
     return [{'forward' : c} for c in check_inputs]
 
+
+def freeze(mod, preserved_attrs : Optional[List[str]] = None):
+    r"""
+    Freezing a :class:`ScriptModule` will clone it and attempt to inline the cloned
+    module's submodules, parameters, and attributes as constants
+    in the TorchScript IR Graph. By default, `forward` will be preserved,
+    as well as attributes specified in `preserved_attrs`. Additionally, attributes which
+    are modified in the `forward` or in a method specified in `preserved_attrs`
+    will be preserved.
+
+    Freezing currently only accepts ScriptModules which are in eval mode.
+
+    Arguments:
+        mod (:class:`ScriptModule`): a module to be frozen
+
+        preserved_attrs (Optional[List[str]]): a list of additional attributes to `forward` to preserve.
```

Comment:
"attributes to `forward`" implies that these are attributes of the forward method, not the module. This isn't correct, right?

I think this might read better as "a list of attributes to preserve in addition to the forward method"

BTW, is there any way to make forward not preserved?

Reply:
there's not an easy way, the use case is very uncommon (there hasn't been any requests for that). you could wrap your module in another module if you absolutely needed to.